### PR TITLE
[chore] Port 2FA tests

### PR DESCRIPTION
**Pull request by battermann**
## Checklist

 - [x] Add a new entry in an appropriate subdirectory of `changelog.d`
 - [x] Read and follow the [PR guidelines](https://docs.wire.com/developer/developer/pr-guidelines.html)

### CODE REVIEW

File: integration/test/Test/Login.hs [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+testLoginVerify6DigitEmailCodeSuccess :: HasCallStack => App ()
+testLoginVerify6DigitEmailCodeSuccess = do
+  (owner, team, []) <- createTeam OwnDomain 0
+  email <- owner %. "email"
+  setTeamFeatureLockStatus owner team "sndFactorPasswordChallenge" "unlocked"
+  setTeamFeatureStatus owner team "sndFactorPasswordChallenge" "enabled"
+  generateVerificationCode owner email
+  code <- getVerificationCode owner "login" >>= getJSON 200 >>= asString
+  bindResponse (loginWith2ndFactor owner email defPassword code) $ \resp -> do
+    resp.status `shouldMatchInt` 200
```
Comment (fisx) [inline integration/test/Test/Login.hs:23]: in the old test, we were checking the json content as well, i think that was useful.

File: integration/test/Test/Login.hs [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+-- @END
+
+-- @SF.Channel @TSFI.RESTfulAPI @S2
+--
+-- Test that login fails with expired second factor email verification code
+testLoginVerify6DigitExpiredCodeFails :: HasCallStack => App ()
+testLoginVerify6DigitExpiredCodeFails = do
+  withModifiedBackend
+    (def {brigCfg = setField "optSettings.setVerificationTimeout" (Aeson.Number 2)})
```
Comment (fisx) [inline integration/test/Test/Login.hs:61]:
```suggestion
    (def {brigCfg = setField "optSettings.setVerificationTimeout" (Aeson.Number 1)})
```

save one second every test run?

(yes, probably an issue with the old test as well, but still.)

File: integration/test/Test/Login.hs [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+testLoginVerify6DigitLimitRetries :: HasCallStack => App ()
+testLoginVerify6DigitLimitRetries = do
+  (owner, team, []) <- createTeam OwnDomain 0
+  email <- owner %. "email"
+  setTeamFeatureLockStatus owner team "sndFactorPasswordChallenge" "unlocked"
+  setTeamFeatureStatus owner team "sndFactorPasswordChallenge" "enabled"
+  generateVerificationCode owner email
+  correctCode <- getVerificationCode owner "login" >>= getJSON 200 >>= asString
+  let wrongCode = "123456"
```
Comment (fisx) [inline integration/test/Test/Login.hs:106]: i guess every one in a million times, this test will probably fail, but that's ok?  :)

**Review by fisx — APPROVED** [review]

File: integration/test/Test/Login.hs (hunk 1 above)
Comment (battermann) [reply integration/test/Test/Login.hs:23]: Can't find it, can you point me to it?

Comment (fisx) [reply integration/test/Test/Login.hs:24]: https://github.com/wireapp/wire-server/pull/3986/files#diff-5d67659df46f564af0bf4bf2cfa2977d5be77b859c66c70cb18fe7c6571b9393L449-L455

Comment (fisx) [reply integration/test/Test/Login.hs:24]: sorry, i was reading it wrong.  all good!

**Pull request merged**